hummingbot/connector/exchange/bittrex/bittrex_api_order_book_data_source.py

```diff
@@ -1,244 +1,186 @@
         try:
-            async with aiohttp.ClientSession() as client:
-                async with client.get(f"{BITTREX_REST_URL}{BITTREX_EXCHANGE_INFO_PATH}", timeout=5) as response:
-                    if response.status == 200:
-                        all_trading_pairs: List[Dict[str, Any]] = await response.json()
-                        return [item["symbol"]
-                                for item in all_trading_pairs
-                                if item["status"] == "ONLINE"]
-        except Exception:
-            # Do nothing if the request fails -- there will be no autocomplete for bittrex trading pairs
-            pass
-        return []
-
-    @staticmethod
-    async def get_snapshot(client: aiohttp.ClientSession, trading_pair: str) -> Dict[str, Any]:
-        # Creates/Reuses connection to obtain a single snapshot of the trading_pair
-        params = {"depth": 25}
-        async with client.get(f"{BITTREX_REST_URL}{BITTREX_EXCHANGE_INFO_PATH}/{trading_pair}/orderbook", params=params) as response:
-            response: aiohttp.ClientResponse = response
-            if response.status != 200:
-                raise IOError(f"Error fetching Bittrex market snapshot for {trading_pair}. "
-                              f"HTTP status is {response.status}.")
-            data: Dict[str, Any] = await response.json()
-            data["sequence"] = response.headers["sequence"]
-            return data
-
-    async def listen_for_subscriptions(self):
-        while True:
-            ws = None
-            try:
-                ws = await self._build_websocket_connection()
-                async for raw_message in self._checked_socket_stream(ws):
-                    decoded: Dict[str, Any] = self._transform_raw_message(raw_message)
-                    self.logger().debug(f"Got ws message {decoded}.")
-                    topic = decoded["type"]
-                    if topic in ["delta", "trade"]:
-                        self._message_queues[topic].put_nowait(decoded)
-            except asyncio.CancelledError:
+            trading_symbols = [await self._connector.exchange_symbol_associated_to_pair(trading_pair=trading_pair)
+                               for trading_pair in self._trading_pairs]
+            subscription_names = [f"trade_{symbol}" for symbol in trading_symbols]
+            subscription_names.extend([f"orderbook_{symbol}_25" for symbol in trading_symbols])
+            response = await self.invoke("Subscribe", [subscription_names])
+            flag = True
+            for i in range(len(subscription_names)):
+                if not response[i]['Success']:
+                    flag = False
+                    print("Subscription to '" + subscription_names[i] + "' failed: " + response[i]["ErrorCode"])
+            if flag:
+                self.logger().info("Subscribed to public order book and trade channels...")
+            else:
                 raise
```

Comment:
The flag variable indicates if all the channels have been subscribed to succesfully. In case one of the channels get a response of unsuccesfuless, an exception is raised to display the log message

Reply:
Added exception with custom description to raise